File: app/src/main/java/br/com/menuvem/lojista/presentation/components/ItemListaCard.kt
package br.com.menuvem.lojista.presentation.components

import androidx.compose.animation.animateColorAsState
import androidx.compose.animation.core.spring
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Delete
import androidx.compose.material.icons.filled.ShoppingCart
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.style.TextDecoration
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import br.com.menuvem.lojista.domain.model.ItemLista
import br.com.menuvem.lojista.domain.model.TendenciaPreco
import java.util.Locale

/**
 * Card de item na lista de compras com:
 * - Checkbox de marcação
 * - Nome e quantidade/unidade
 * - Preço e subtotal (editável)
 * - Indicador de tendência (se o item estiver vinculado a insumo)
 * - Tachado quando comprado
 */
@Composable
fun ItemListaCard(
    item: ItemLista,
    tendencia: TendenciaPreco?,
    onToggle: (Boolean, Double) -> Unit,
    onDelete: () -> Unit,
    modifier: Modifier = Modifier
) {
    var preco by remember(item.id, item.precoUnitario) {
        mutableStateOf(if (item.precoUnitario > 0) item.precoUnitario.toString() else "")
    }
    var showDeleteDialog by remember { mutableStateOf(false) }

    val bgColor by animateColorAsState(
        targetValue = if (item.comprado)
            MaterialTheme.colorScheme.surfaceVariant.copy(alpha = 0.6f)
        else
            MaterialTheme.colorScheme.surface,
        animationSpec = spring(),
        label = "bg"
    )

    if (showDeleteDialog) {
        AlertDialog(
            onDismissRequest = { showDeleteDialog = false },
            title = { Text("Remover item?") },
            text = { Text("\"${item.nomeItem}\" será removido da lista.") },
            confirmButton = {
                TextButton(onClick = {
                    onDelete()
                    showDeleteDialog = false
                }) { Text("Remover", color = MaterialTheme.colorScheme.error) }
            },
            dismissButton = {
                TextButton(onClick = { showDeleteDialog = false }) { Text("Cancelar") }
            }
        )
    }

    Card(
        modifier = modifier.fillMaxWidth(),
        colors = CardDefaults.cardColors(containerColor = bgColor),
        elevation = CardDefaults.cardElevation(defaultElevation = if (item.comprado) 0.dp else 2.dp)
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 12.dp, vertical = 10.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            // Checkbox
            Checkbox(
                checked = item.comprado,
                onCheckedChange = { checked ->
                    val precoFinal = preco.replace(",", ".").toDoubleOrNull() ?: 0.0
                    onToggle(checked, precoFinal)
                },
                colors = CheckboxDefaults.colors(
                    checkedColor = MaterialTheme.colorScheme.primary,
                    checkmarkColor = Color.White
                )
            )

            Spacer(modifier = Modifier.width(8.dp))

            // Conteúdo principal
            Column(modifier = Modifier.weight(1f)) {
                // Nome + badge de insumo + tendência
                Row(
                    verticalAlignment = Alignment.CenterVertically,
                    horizontalArrangement = Arrangement.spacedBy(6.dp)
                ) {
                    Text(
                        text = item.nomeItem,
                        style = MaterialTheme.typography.titleSmall,
                        textDecoration = if (item.comprado) TextDecoration.LineThrough else TextDecoration.None,
                        color = if (item.comprado)
                            MaterialTheme.colorScheme.onSurfaceVariant
                        else
                            MaterialTheme.colorScheme.onSurface,
                        maxLines = 1,
                        overflow = TextOverflow.Ellipsis,
                        modifier = Modifier.weight(1f, fill = false)
                    )

                    if (item.estaVinculado) {
                        Surface(
                            shape = CircleShape,
                            color = MaterialTheme.colorScheme.primaryContainer
                        ) {
                            Icon(
                                imageVector = Icons.Default.ShoppingCart,
                                contentDescription = "Insumo cadastrado",
                                tint = MaterialTheme.colorScheme.primary,
                                modifier = Modifier
                                    .padding(3.dp)
                                    .size(10.dp)
                            )
                        }
                    }

                    tendencia?.let {
                        TendenciaIndicador(tendencia = it)
                    }
                }

                Spacer(modifier = Modifier.height(2.dp))

                // Quantidade e unidade
                Text(
                    text = "${formatarNumero(item.quantidade)} ${item.unidade}",
                    style = MaterialTheme.typography.bodySmall,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
            }

            Spacer(modifier = Modifier.width(8.dp))

            // Preço e subtotal
            Column(horizontalAlignment = Alignment.End) {
                if (!item.comprado) {
                    // Campo editável de preço
                    OutlinedTextField(
                        value = preco,
                        onValueChange = { preco = it.filter { c -> c.isDigit() || c == '.' || c == ',' } },
                        label = { Text("R$", style = MaterialTheme.typography.labelSmall) },
                        modifier = Modifier.width(90.dp),
                        singleLine = true,
                        textStyle = MaterialTheme.typography.bodySmall,
                        colors = OutlinedTextFieldDefaults.colors(
                            focusedBorderColor = MaterialTheme.colorScheme.primary,
                            unfocusedBorderColor = MaterialTheme.colorScheme.outline.copy(alpha = 0.5f)
                        )
                    )
                } else {
                    // Preço final
                    Text(
                        text = "R$ ${formatarMoeda(item.precoUnitario)}",
                        style = MaterialTheme.typography.bodySmall,
                        color = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                    Text(
                        text = "= R$ ${formatarMoeda(item.precoTotal)}",
                        style = MaterialTheme.typography.labelMedium,
                        color = MaterialTheme.colorScheme.primary
                    )
                }
            }

            // Botão deletar
            IconButton(
                onClick = { showDeleteDialog = true },
                modifier = Modifier.size(32.dp)
            ) {
                Icon(
                    imageVector = Icons.Default.Delete,
                    contentDescription = "Remover",
                    tint = MaterialTheme.colorScheme.error.copy(alpha = 0.6f),
                    modifier = Modifier.size(18.dp)
                )
            }
        }
    }
}

private fun formatarNumero(value: Double): String =
    if (value == value.toLong().toDouble()) value.toLong().toString()
    else String.format(Locale("pt", "BR"), "%.2f", value)

fun formatarMoeda(value: Double): String =
    String.format(Locale("pt", "BR"), "%.2f", value)

File: lib/presentation/shopping/item_lista_card.dart
import 'package:flutter/material.dart';

import '../../domain/model/item_lista.dart';
import '../../domain/model/tendencia_preco.dart';
import '../components/formatters.dart';
import '../components/tendencia_indicador.dart';

/// Card de item na lista de compras: checkbox, nome, quantidade/unidade,
/// preço editável (subtotal quando comprado), tendência e exclusão.
class ItemListaCard extends StatefulWidget {
  const ItemListaCard({
    super.key,
    required this.item,
    required this.tendencia,
    required this.onToggle,
    required this.onDelete,
  });

  final ItemLista item;
  final TendenciaPreco? tendencia;
  final void Function(bool comprado, double preco) onToggle;
  final VoidCallback onDelete;

  @override
  State<ItemListaCard> createState() => _ItemListaCardState();
}

class _ItemListaCardState extends State<ItemListaCard> {
  late final _precoController = TextEditingController(
    text: widget.item.precoUnitario > 0 ? widget.item.precoUnitario.toString() : '',
  );

  @override
  void dispose() {
    _precoController.dispose();
    super.dispose();
  }

  static String _formatarNumero(double value) {
    return value == value.truncateToDouble()
        ? value.toInt().toString()
        : value.toStringAsFixed(2);
  }

  @override
  Widget build(BuildContext context) {
    final item = widget.item;
    final colorScheme = Theme.of(context).colorScheme;

    return Card(
      color: item.comprado ? colorScheme.surfaceContainerHighest.withValues(alpha: 0.6) : null,
      elevation: item.comprado ? 0 : 2,
      child: Padding(
        padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 10),
        child: Row(
          children: [
            Checkbox(
              value: item.comprado,
              onChanged: (checked) {
                final preco = parseDecimalPtBr(_precoController.text) ?? 0.0;
                widget.onToggle(checked ?? false, preco);
              },
            ),
            const SizedBox(width: 8),
            Expanded(
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Row(
                    children: [
                      Flexible(
                        child: Text(
                          item.nomeItem,
                          maxLines: 1,
                          overflow: TextOverflow.ellipsis,
                          style: Theme.of(context).textTheme.titleSmall?.copyWith(
                                decoration: item.comprado
                                    ? TextDecoration.lineThrough
                                    : null,
                                color: item.comprado
                                    ? colorScheme.onSurfaceVariant
                                    : colorScheme.onSurface,
                              ),
                        ),
                      ),
                      if (item.estaVinculado) ...[
                        const SizedBox(width: 6),
                        Container(
                          padding: const EdgeInsets.all(3),
                          decoration: BoxDecoration(
                            color: colorScheme.primaryContainer,
                            shape: BoxShape.circle,
                          ),
                          child: Icon(
                            Icons.shopping_cart,
                            size: 10,
                            color: colorScheme.primary,
                          ),
                        ),
                      ],
                      if (widget.tendencia != null) ...[
                        const SizedBox(width: 6),
                        TendenciaIndicador(tendencia: widget.tendencia!),
                      ],
                    ],
                  ),
                  const SizedBox(height: 2),
                  Text(
                    '${_formatarNumero(item.quantidade)} ${item.unidade}',
                    style: Theme.of(context)
                        .textTheme
                        .bodySmall
                        ?.copyWith(color: colorScheme.onSurfaceVariant),
                  ),
                ],
              ),
            ),
            const SizedBox(width: 8),
            if (!item.comprado)
              SizedBox(
                width: 90,
                child: TextField(
                  controller: _precoController,
                  decoration: const InputDecoration(labelText: 'R\$', isDense: true),
                  style: Theme.of(context).textTheme.bodySmall,
                  keyboardType: const TextInputType.numberWithOptions(decimal: true),
                )
              )
            else
              Column(
                crossAxisAlignment: CrossAxisAlignment.end,
                children: [
                  Text(
                    'R\$ ${formatarMoeda(item.precoUnitario)}',
                    style: Theme.of(context)
                        .textTheme
                        .bodySmall
                        ?.copyWith(color: colorScheme.onSurfaceVariant),
                  ),
                  Text(
                    '= R\$ ${formatarMoeda(item.precoTotal)}',
                    style: Theme.of(context)
                        .textTheme
                        .labelMedium
                        ?.copyWith(color: colorScheme.primary),
                  ),
                ],
              ),
            IconButton(
              onPressed: () => _confirmDelete(context),
              icon: Icon(
                Icons.delete,
                size: 18,
                color: colorScheme.error.withValues(alpha: 0.6),
              ),
            ),
          ],
        ),
      ),
    );
  }

  void _confirmDelete(BuildContext context) {
    showDialog<void>(
      context: context,
      builder: (context) => AlertDialog(
        title: const Text('Remover item?'),
        content: Text('"${widget.item.nomeItem}" será removido da lista.'),
        actions: [
          TextButton(
            onPressed: () => Navigator.of(context).pop(),
            child: const Text('Cancelar'),
          ),
          TextButton(
            onPressed: () {
              Navigator.of(context).pop();
              widget.onDelete();
            },
            child: Text('Remover', style: TextStyle(color: Theme.of(context).colorScheme.error)),
          ),
        ],
      ),
    );
  }
}
